x-pack/legacy/plugins/lens/public/xy_visualization_plugin/to_expression.ts

```diff
@@ -55,12 +56,55 @@ export const toExpression = (state: State, frame: FramePublicAPI): Ast | null =>
     }),
   };
 
-  return buildExpression(stateWithValidAccessors, xyTitles(state.layers[0], frame), frame);
+  const metadata: Record<string, Record<string, OperationMetadata | null>> = {};
+  state.layers.forEach(layer => {
+    metadata[layer.layerId] = {};
+    const datasource = frame.datasourceLayers[layer.layerId];
+    datasource.getTableSpec().forEach(column => {
+      const operation = frame.datasourceLayers[layer.layerId].getOperationForColumnId(
+        column.columnId
+      );
+      metadata[layer.layerId][column.columnId] = operation;
+    });
+  });
+
+  return buildExpression(
+    stateWithValidAccessors,
+    xyTitles(state.layers[0], frame),
+    metadata,
+    frame
+  );
 };
 
+export function getScaleType(metadata: OperationMetadata | null, defaultScale: ScaleType) {
+  if (!metadata) {
+    return defaultScale;
+  }
+
+  // use scale information if available
+  if (metadata.scale === 'ordinal') {
+    return ScaleType.Ordinal;
+  }
+  if (metadata.scale === 'interval' || metadata.scale === 'ratio') {
+    return metadata.dataType === 'date' ? ScaleType.Time : ScaleType.Linear;
+  }
+
+  // fall back to data type if necessary
+  switch (metadata.dataType) {
+    case 'boolean':
+    case 'string':
+      return ScaleType.Ordinal;
+    case 'date':
+      return ScaleType.Time;
+    default:
+      return ScaleType.Linear;
+  }
```

Comment:
I don't see this using `metadata.isHistogram`?

Reply:
It's passed in as a separate property in line 169. I tried to stick closely to the elastic-charts API when possible.